symbolic-symcache/src/new/writer.rs

```diff
@@ -308,6 +315,111 @@ impl SymCacheConverter {
         }
     }
 
+    #[cfg(feature = "il2cpp")]
+    /// Processes a set of [`UsymSymbols`], passing all mapped symbols into the converter.
+    pub fn process_usym(&mut self, usym: &UsymSymbols) -> Result<(), SymCacheError> {
+        // Assume records they are sorted by address; There's a test that guarantees this
+
+        let mapped_records = usym.records().filter_map(|r| match r {
+            UsymSourceRecord::Unmapped(_) => None,
+            UsymSourceRecord::Mapped(r) => Some(r),
+        });
+
+        let mut curr_id: Vec<Cow<'_, str>> = Vec::default();
```

Comment:
since this whole discussion started with "don’t allocate in tight loops", using a `Vec` still does. I did suggest a tuple, or even a `Option<(Cow,Cow)>` since the first one is empty, that would avoid allocating a vec ;-)

Reply:
i didn't realize that tuples worked that way under the hood, thanks for the suggestion!